test-result-summary-client/src/PerfCompare/PerfCompare.jsx

```diff
@@ -168,6 +170,79 @@ export default class PerfCompare extends Component {
         });
     };
 
+    handleGetAzDoRuns = async () => {
+        this.setState({
+            submitStatus: 'submit',
+            progress: 20,
+            progressStatus: 'active',
+        });
+
+        const baselineTestResultsJson = await fetchData(
+            `/api/getAzDoRun?buildId=${this.state.selectedRuns.baselineID}`
+        );
+
+        const testTestResultsJson = await fetchData(
+            `/api/getAzDoRun?buildId=${this.state.selectedRuns.testID}`
+        );
+
+        // Check if the given builds are valid
+        let displayErrorMessage = '';
+        let hasData = false;
+        if (!baselineTestResultsJson || !baselineTestResultsJson.length > 0) {
+            displayErrorMessage = 'Baseline build not found. ';
+        } else {
+            hasData = true;
+        }
+        if (!testTestResultsJson || !testTestResultsJson.length > 0) {
+            displayErrorMessage += 'Test build not found';
+        } else {
+            hasData = true;
+        }
+        // Data received is not valid
+        if (!hasData) {
+            this.setState({
+                inputURL: {
+                    baselineID: '',
+                    testID: '',
+                },
+                selectedRuns: {
+                    baselineID: '',
+                    testID: '',
+                },
+                submitStatus: 'none',
+                progress: 0,
+                progressStatus: '',
+                displayAlert: {
+                    status: true,
+                    message: displayErrorMessage,
+                },
+            });
+            return;
+        }
+
+        let baselineRunJSON = new AzDoRunJSON(baselineTestResultsJson);
+        let testRunJSON = new AzDoRunJSON(testTestResultsJson);
+
+        baselineRunJSON.init(() => {
+            this.setState({
+                progress: 40,
+                progressStatus: 'active',
+            });
+
+            testRunJSON.init(async () => {
+                this.setState({
+                    benchmarkRuns: {
+                        benchmarkRunBaseline: baselineRunJSON,
+                        benchmarkRunTest: testRunJSON,
+                    },
+                    progress: 60,
```

Comment:
parameter that should be passed in?

Reply:
no this just updates a progress bar in the UI